app/src/main/res/layout/screen_patients.xml

```diff
@@ -7,68 +7,70 @@
-    android:scaleType="fitXY"
-    android:src="@drawable/illustrations_homescreen"
-    tools:ignore="ContentDescription" />
+      <org.simple.clinic.widgets.OmegaCenterIconButton
+        android:id="@+id/patients_search_patients"
+        android:layout_width="match_parent"
+        android:layout_height="56dp"
+        android:layout_marginEnd="@dimen/spacing_16"
+        android:layout_marginStart="@dimen/spacing_16"
+        android:layout_marginTop="@dimen/spacing_16"
+        android:background="@drawable/button_outline_ripple"
+        android:drawablePadding="@dimen/spacing_8"
+        android:drawableStart="@drawable/ic_search_24dp"
+        android:fontFamily="sans-serif"
+        android:gravity="center"
+        android:paddingStart="20dp"
+        android:text="@string/home_search_patient_name_phone"
+        android:textAllCaps="false"
+        android:textColor="?colorAccent"
+        android:textSize="@dimen/textsize_18"
+        app:drawableTint="?colorAccent"
+        tools:ignore="MissingPrefix,RtlSymmetry" />
 
-  <FrameLayout
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:layout_below="@id/patients_record_bp_illustration">
+      <org.simple.clinic.widgets.OmegaCenterIconButton
+        android:id="@+id/patients_scan_simple_card"
+        style="@style/Clinic.V2.Button.WithIcon.Blue1"
+        android:layout_width="match_parent"
+        android:layout_height="56dp"
+        android:layout_below="@id/patients_search_patients"
+        android:layout_marginEnd="@dimen/spacing_16"
+        android:layout_marginStart="@dimen/spacing_16"
+        android:layout_marginTop="@dimen/spacing_16"
+        android:drawableStart="@drawable/patient_id_card"
+        android:gravity="center"
+        android:text="@string/patients_scan_simple_card"
+        android:textAllCaps="false"
+        android:textAppearance="@style/Clinic.V2.TextAppearance.Body0Left.White100"
+        app:drawableTint="@color/white100"
+        tools:ignore="MissingPrefix,RtlSymmetry" />
 
-    <TextView
-      style="@style/Clinic.V2.TextAppearance.Subtitle1Center.Grey1"
-      android:layout_width="match_parent"
-      android:layout_height="wrap_content"
-      android:layout_gravity="center"
-      android:layout_marginEnd="@dimen/spacing_16"
-      android:layout_marginStart="@dimen/spacing_16"
-      android:text="@string/patients_record_every_patient" />
-  </FrameLayout>
+      <ImageView
+        android:id="@+id/patients_record_bp_illustration"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/patients_scan_simple_card"
+        android:layout_centerHorizontal="true"
+        android:adjustViewBounds="true"
+        android:scaleType="fitXY"
+        android:src="@drawable/illustrations_homescreen"
+        tools:ignore="ContentDescription" />
+
+      <org.simple.clinic.patient.recent.RecentPatientsView
```

Comment:
There are two major problems here.

1. Having nested scrollable views that scroll in the same direction/axis is never a good idea.

2. A `RecyclerView` with height set to `wrap_content` is as good as a `LinearLayout` because there's no recycling happening here. 

A `RecyclerView` is an extremely complex layout that is built to specially handle recycling of items in a performant manner using various optimizations . By using `LinearLayout`, you're forcing RV to perform those optimizations and end up wasting them. Plus, if I see a `RecyclerView`, I expect things to recycle. I'm not sure how to put this in better words, but it's _wrong_.

Just use a `LinearLayout` and plug it with your adapter. Even if you plan to reuse this View in multiple screens, you're actually only going to reuse the adapter. There's no advantage of reusing a `RecyclerView` in two places.

Using a LinearLayout with an adapter requires just 4 lines. 

```kotlin
recentPatientItems.map { item ->
  val viewHolder = adapter.onCreateViewHolder(...)
  linearLayout.addView(viewHolder.itemView)
  viewHolder.render(item)
}
```

Reply:
I was thinking that it's a good idea that you want to reuse one View in both `patients screen` and `full list of recent patients` screen, but I think it's okay to duplicate some logic. I'm sure in the future we'll get different kind of requirements for behaviors in both the screens and you'll find better about it. 

This has already happened with facility selection in the past. There's one screen that's used during nurse registration and one post login (from home screen). They had exactly the same behavior in the beginning, but they diverged later. They're merging again now, but my point is that it's okay to not reuse recent patients list.

Reply:
As discussed in person, we'll keep this as RecyclerView itself.

Reply:
^ that's not what we discussed in person, but it's okay if we're in disagreement.

Reply:
My reasons for keeping this: it's more idiomatic to use a component that's built for templatized rows, i.e., RecyclerView, instead of shoehorning `LinearLayout` and making it work with RecyclerView's adapters.

There's no degradation of performance as you've stated. Once we state RecyclerView's height as wrap_content, it does not try to needlessly recycle the views. All the child views are created in one shot when the view is created.

Also, if we start using LinearLayout and RecyclerView for the same purposes, we lose consistency across the project. Breaking this does not make sense for this use case as recyclerview can handle all those scenarios.

I realize the change to LinearLayout is just addition of couple of lines, but this breaks the obvious way of implementing an obvious UI component.

Reply:
Also, this is what I thought we had discussed in person. I am sorry, if I didn't convey this properly.